core/modules/replica/Common.h

```diff
@@ -135,6 +138,35 @@ class SqlColDef {
     std::string type;
 };
 
+
+/**
+ * This class is an abstraction for columns within table index
+ * specifications.
+ */
+class SqlIndexColumn {
+public:
+
+    SqlIndexColumn() = default;
+    SqlIndexColumn(std::string const name_,
+                   size_t length_,
+                   bool ascending_)
+        :   name(name_),
+            length(length_),
+            ascending(ascending_) {
+    }
+
+    SqlIndexColumn(SqlIndexColumn const&) = default;
+    SqlIndexColumn& operator=(SqlIndexColumn const&) = default;
+
+    ~SqlIndexColumn() = default;
+
+
+    std::string name;
+    size_t length = 0;
+    bool ascending = true;
```

Comment:
make private with accessors?

Reply:
What you're saying is an option. Though, that would be definitively an "overkill" for this tiny class, which in essence, is just a `struct` with a constructor (handy for things like `std:vector<T>::emplace_back()`).